mlflow/utils/virtualenv.py

```diff
@@ -0,0 +1,254 @@
+    _logger.info("Installing python %s", version)
+    # pyenv-win doesn't support `--skip-existing` but its behavior is enabled by default
+    # https://github.com/pyenv-win/pyenv-win/pull/314
+    pyenv_install_options = ("--skip-existing",) if _IS_UNIX else ()
+    _exec_cmd(["pyenv", "install", *pyenv_install_options, version], capture_output=False)
+
+    if _IS_UNIX:
+        pyenv_root = _exec_cmd(["pyenv", "root"], capture_output=True).stdout.strip()
+        path_to_bin = ("bin", "python")
+    else:
+        # pyenv-win doesn't provide the `pyenv root` command
+        pyenv_root = os.getenv("PYENV_ROOT")
+        if pyenv_root is None:
+            raise MlflowException("Environment variable 'PYENV_ROOT' must be set")
+        path_to_bin = ("python.exe",)
+    return Path(pyenv_root).joinpath("versions", version, *path_to_bin)
+
+
+def _get_python_env(local_model_path):
+    """
+    Constructs `PythonEnv` from the model artifacts stored in `local_model_path`. If
+    `python_env.yaml` is available, use it, otherwise extract model dependencies from `conda.yaml`.
+    If `conda.yaml` contains conda dependencies except `python`, `pip`, `setuptools`, and, `wheel`,
+    an `MlflowException` is thrown because conda dependencies cannot be installed in a virtualenv
+    environment.
+
+    :param local_model_path: Local directory containing the model artifacts.
+    :return: `PythonEnv` instance.
+    """
+    python_env_file = local_model_path / _PYTHON_ENV_FILE_NAME
+    requirements_file = local_model_path / _REQUIREMENTS_FILE_NAME
+    conda_env_file = local_model_path / _CONDA_ENV_FILE_NAME
+    if python_env_file.exists():
+        return PythonEnv.from_yaml(python_env_file)
+    else:
+        _logger.info(
+            "This model is missing %s, which is because it was logged in an older version"
+            "of MLflow (< 1.26.0) that does not support restoring a model environment with "
+            "virtualenv. Attempting to extract model dependencies from %s and %s instead.",
+            _PYTHON_ENV_FILE_NAME,
+            _REQUIREMENTS_FILE_NAME,
+            _CONDA_ENV_FILE_NAME,
+        )
+        conda_deps = _get_conda_dependencies(conda_env_file)
+        build_packages = ("python", *PythonEnv.BUILD_PACKAGES)
+        conda_deps = [d for d in conda_deps if _get_package_name(d) not in build_packages]
+        if conda_deps:
+            raise MlflowException(
+                f"Cannot restore this model's environment with virtualenv because it contains "
+                f"conda dependencies: {conda_deps}."
+            )
+        if requirements_file.exists():
+            deps = PythonEnv.get_dependencies_from_conda_yaml(conda_env_file)
+            deps.pop("dependencies", None)
+            return PythonEnv(
+                **deps,
+                dependencies=[f"-r {_REQUIREMENTS_FILE_NAME}"],
+            )
+        else:
+            return PythonEnv.from_conda_yaml(conda_env_file)
```

Comment:
@dbczumar Code that relates to https://github.com/mlflow/mlflow/pull/5380#discussion_r842424627

Reply:
```
return PythonEnv(
                **deps,
                dependencies=[f"-r {_REQUIREMENTS_FILE_NAME}"],
            )
```

The **deps already includes a "dependencies" key, does it conflicts with the folllowing `dependencies=` ?

Using
```
return PythonEnv(
                python=deps.python,
                build_dependencies=deps.build_dependencies,
                dependencies=[f"-r {_REQUIREMENTS_FILE_NAME}"],
            )
```
might be more clear.

Reply:
I think you missed line 156

Reply:
Applied your suggestion.